hail/python/hailtop/auth/tokens.py

```diff
@@ -64,12 +63,15 @@ def write(self):
             json.dump(self._tokens, f)
 
 
-tokens = None
+tokens = {}
 
 
-def get_tokens():
+def get_tokens(file=None):
     global tokens
 
-    if not tokens:
-        tokens = Tokens()
-    return tokens
+    if file is None:
```

Comment:
I had it like this because I don't think tokens is a dictionary of file:{tokens}. 

```
class Tokens(collections.abc.MutableMapping):
    def __init__(self, tokens_file=None):
        if tokens_file is None:
            tokens_file = self.get_tokens_file()
        if os.path.isfile(tokens_file):
            with open(tokens_file, 'r') as f:
                self._tokens = json.load(f)
            log.info(f'tokens loaded from {tokens_file}')
        else:
            log.info(f'tokens file not found: {tokens_file}')
            self._tokens = {}

def get_tokens(tokens_file=None):
    global tokens

    if not tokens:
        tokens = Tokens(tokens_file=tokens_file)
    return tokens
```

Reply:
It's not currently! But I need to be able to call `get_tokens` for multiple different token files in the same python environment, so I changed it to be a dictionary instead of a single token. The use case is something like 
```
dev_tokens=get_tokens(dev_file)
tokens = get_tokens(normal_file)
```

which doesn't return two separate tokens if I'm only caching the first file.